tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,332 @@
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes = util.Call(t, "eth_getBlockByNumber", []interface{}{"pending", true})
+	var postTxPendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &postTxPendingBlock)
+	require.NoError(t, err)
+	postTxPendingTxs := len(postTxPendingBlock["transactions"].([]interface{}))
+	require.Greater(t, postTxPendingTxs, preTxPendingTxs)
+
+	rpcRes = util.Call(t, "eth_getBlockByNumber", []interface{}{"latest", true})
+	var postTxLatestBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &postTxLatestBlock)
+	require.NoError(t, err)
+	postTxLatestTxs := len(postTxLatestBlock["transactions"].([]interface{}))
+	require.Equal(t, preTxLatestTxs, postTxLatestTxs)
+
+	require.Greater(t, postTxPendingTxs, preTxPendingTxs)
+}
+
+func TestEth_Pending_GetTransactionByBlockNumberAndIndex(t *testing.T) {
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes := util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"pending", "0x1"})
+	var pendingBlock map[string]interface{}
+	err := json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	require.Equal(t, pendingBlock["blockHash"], nil)
+	require.Equal(t, pendingBlock["blockNumber"], nil)
+	require.Equal(t, pendingBlock["transactionIndex"], nil)
+	require.NotEmpty(t, pendingBlock["hash"])
+
+	rpcRes = util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"latest", "0x1"})
+	var latestBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &latestBlock)
+	require.NoError(t, err)
+
+	require.Empty(t, latestBlock)
+}
+
+func TestEth_Pending_GetTransactionByHash(t *testing.T) {
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	txRes := util.Call(t, "eth_sendTransaction", param)
+	var txHash common.Hash
+	err := txHash.UnmarshalJSON(txRes.Result)
+	require.NoError(t, err)
+
+	rpcRes := util.Call(t, "eth_getTransactionByHash", []interface{}{txHash})
+	var pendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, pendingBlock)
+	require.Equal(t, nil, pendingBlock["blockHash"])
+	require.Equal(t, nil, pendingBlock["blockNumber"])
+	require.Equal(t, nil, pendingBlock["transactionIndex"])
+	require.NotEmpty(t, pendingBlock["hash"])
+	require.NotEmpty(t, pendingBlock["value"], "0xa")
```

Comment:
same here, i thought `eth_getTransactionByHash` returned a transaction?

Reply:
does this look okay now?